CRD generator fails with `ClassCastException` in some cases

With some annotated classes, `mvn` failed compilation with an abstract
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project: Fatal error compiling: java.lang.ClassCastException: class java.util.ArrayList cannot be cast to class java.lang.String (java.util.ArrayList and java.lang.String are in module java.base of loader 'bootstrap') -> [Help 1]
```
error

On further investigation, this is failing in the CRD generator and is a regression in 5.7.1 from https://github.com/fabric8io/kubernetes-client/issues/3431: if fields have any annotations with non-string `value` attributes, it will fail since the code always tries to cast it to a `String`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,6 +12,7 @@
 
 #### Bugs
 * Fix: Extension annotator doesn't generate XxxEditable classes
+* Fix #3450: CRD generator fails with ClassCastException in some cases
 
 #### Dependency Upgrade
 * Fix #3438: Upgrade Sundrio to 0.50.1

--- a/crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java
+++ b/crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java
@@ -190,19 +190,20 @@ public abstract class AbstractJsonSchema<T, B> {
 
     public void process() {
       annotations.forEach(a -> {
-        final String fromAnnotation = (String) a.getParameters().get("value");
         switch (a.getClassRef().getFullyQualifiedName()) {
           case ANNOTATION_NOT_NULL:
             required = true;
             break;
           case ANNOTATION_JSON_PROPERTY:
-            if (!Strings.isNullOrEmpty(fromAnnotation) && !propertyName.equals(fromAnnotation)) {
-              renamedTo = fromAnnotation;
+            final String nameFromAnnotation = (String) a.getParameters().get("value");
+            if (!Strings.isNullOrEmpty(nameFromAnnotation) && !propertyName.equals(nameFromAnnotation)) {
+              renamedTo = nameFromAnnotation;
             }
             break;
           case ANNOTATION_JSON_PROPERTY_DESCRIPTION:
-            if (!Strings.isNullOrEmpty(fromAnnotation)) {
-              description = fromAnnotation;
+            final String descriptionFromAnnotation = (String) a.getParameters().get("value");
+            if (!Strings.isNullOrEmpty(descriptionFromAnnotation)) {
+              description = descriptionFromAnnotation;
             }
             break;
         }

--- a/crd-generator/api/src/test/java/io/fabric8/crd/example/annotated/AnnotatedSpec.java
+++ b/crd-generator/api/src/test/java/io/fabric8/crd/example/annotated/AnnotatedSpec.java
@@ -18,6 +18,7 @@ package io.fabric8.crd.example.annotated;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.annotation.JsonPropertyDescription;
 
+import javax.validation.constraints.Min;
 import javax.validation.constraints.NotNull;
 
 public class AnnotatedSpec {
@@ -30,6 +31,8 @@ public class AnnotatedSpec {
   @NotNull
   private boolean emptySetter;
   private AnnotatedEnum anEnum;
+  @Min(0) // a non-string value attribute
+  private int sizedField;
 
   @JsonProperty("from-getter")
   @JsonPropertyDescription("from-getter-description")

--- a/crd-generator/api/src/test/java/io/fabric8/crd/generator/v1/JsonSchemaTest.java
+++ b/crd-generator/api/src/test/java/io/fabric8/crd/generator/v1/JsonSchemaTest.java
@@ -75,7 +75,7 @@ class JsonSchemaTest {
     assertEquals(2, properties.size());
     final JSONSchemaProps specSchema = properties.get("spec");
     Map<String, JSONSchemaProps> spec = specSchema.getProperties();
-    assertEquals(5, spec.size());
+    assertEquals(6, spec.size());
 
     // check descriptions are present
     assertTrue(spec.containsKey("from-field"));

